src/telemetry/telemetry.c

```diff
@@ -60,27 +73,19 @@ telemetry_parse_version(const char *json, VersionInfo *installed_version, Versio
 
 	result->versionstr = text_to_cstring(DatumGetTextPP(version));
 
-	result->is_up_to_date = false;
-
 	if (result->versionstr == NULL)
 	{
 		result->errhint = "no version string in response";
 		return false;
 	}
 
-	/*
-	 * Now parse the version string. We expect format to be
-	 * XX.XX.XX-<prerelease_tag>, and if not, we error out
-	 */
-	if (!version_parse(result->versionstr, &result->vinfo))
+	for (i = 0; i < NUM_VERSION_CHARS_TO_PRINT && i < strlen(result->versionstr); i++)
```

Comment:
This calls `strlen` each iteration, while it won't matter for reasonable strings, we should probably pull it out of the loop, so the look isn't quadratic.

Reply:
What is the purpose of `NUM_VERSION_CHARS_TO_PRINT`?

Is it kind of a "rough" estimate of the max length of a valid version string?

Given that we don't know here what an actual valid version would look like, then I am not sure it is worth doing any validation at all. I think we can just print whatever the server returns as long as it doesn't pose a security risk.

Reply:
@erimatnor How about (taking into account your comment below as well) eliminating `NUM_VERSION_CHARS_TO_PRINT`, but still validating the entire string (i.e. make sure every char is either alphanumeric or '.'). If there is an invalid digit, print a warning, otherwise print the version string?

Reply:
Sounds good to me.